src/datahike/tools.cljc

```diff
@@ -209,3 +234,42 @@
     ([dst x]
      (let [k (f x)]
        (assoc! dst k (conj (get dst k []) x))))))
+
+(defn range-subset-tree
```

Comment:
This function only seems to be used in its test. Do we still need it? It would be good to know what it is for then.

Reply:
It is used in `datahike.query/substitution-expansion` and `datahike.query/make-vec-lookup-ref-replacer`. I think you may have missed it because the diff of `query.cljc` is too big to be displayed by default in Github.

Reply:
Thank you for pointing that out!